test_unstructured/file_utils/test_filetype.py

```diff
@@ -90,6 +90,7 @@ def test_it_detects_correct_file_type_for_CFB_and_ZIP_subtypes_detected_by_direc
         (FileType.WAV, "CantinaBand3.wav", "audio/wav"),
         (FileType.XML, "factbook.xml", "application/xml"),
         (FileType.ZIP, "simple.zip", "application/zip"),
+        (FileType.NDJSON, "spring-weather.html.ndjson", "application/x-ndjson"),
```

Comment:
I would add a couple special-case tests below in the special-cases section, not so much for present purposes as to defend against later adjustments/rearrangements of ordering.

Cases I would try:
- correctly identifies when no filename (so no extension) but correct content-type is specified.
- correct when correct extension is specified by wrong content-type (especially `application/json`)
- correct when content-type is right but extension is `.json`.

Not sure if all those can be made to work without creating a tangle, but worth a try I expect, should only take a few minutes to give those a whirl.

Reply:
I would like to keep the changes in this PR isolated to only the ndjson support. I don't think the special cases you're asking about would really fall under that?

Reply:
The tests I'm proposing are all specifically for NDJSON, like the first would be that an NDJSON file is correctly identified as `FileType.NDJSON` when no filename is provided but `application/x-ndjson` content-type is provided.

The ordering of filetype-detection tests is subject to change as special cases arise, so it's important to have all the cases defended against someone inadvertently breaking your code by another change to say get CSV properly identified in some special case.

This is the kind of test I'm talking about:
```python

def test_it_identifies_NDJSON_for_file_like_object_with_no_name_but_NDJSON_content_type():
    with open(example_doc_path("simple.ndjson"), "rb") as f:
        file = io.BytesIO(f.read())
    assert detect_filetype(file=file, content_type=FileType.NDJSON.mime_type) == FileType.NDJSON

def test_it_identifies_NDJSON_for_file_with_ndjson_extension_but_JSON_content_type():
    file_path = example_doc_path("simple.ndjson")
    assert detect_filetype(file_path, content_type=FileType.JSON.mime_type) == FileType.NDJSON
```